pipeline.cpp

```diff
@@ -74,14 +74,65 @@ void DrawLine(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* cons
     // Your code goes here
 }
 
+/**********************************************************************
+ * CrossProduct
+ * computes the cross product (also the determinant) of a 2X2 matrix
+ **********************************************************************/
+float CrossProduct(Vertex const v1, Vertex const v2)
+{
+    return ((v1.x * v2.y) - (v1.y * v2.x));
+}
+
 /*************************************************************
  * DRAW_TRIANGLE
  * Renders a triangle to the target buffer. Essential 
  * building block for most of drawing.
  ************************************************************/
 void DrawTriangle(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* const attrs, Attributes* const uniforms, FragmentShader* const frag)
 {
-    // Your code goes here
+    //create the bounding box with the max and min x, y values
+    float maxX = MAX3(triangle[0].x, triangle[1].x, triangle[2].x);
```

Comment:
Does this do the same thing as the equations the readings showed us?

Reply:
Yeah this is defined for us in the definitions.h, he just saved us from having to import a library.